pkg/resource/resource.go

```diff
@@ -43,6 +43,15 @@ const (
 	AnnotationDelimiter = "/"
 )
 
+// External resources are tagged/labelled with the following keys in the cloud
+// provider API if the type supports.
+const (
+	ExternalResourceKindTagKey     = "crossplane.io/kind"
+	ExternalResourceNameTagKey     = "crossplane.io/name"
+	ExternalResourceClassTagKey    = "crossplane.io/class"
+	ExternalResourceProviderTagKey = "crossplane.io/provider"
```

Comment:
Nit: I think we have two formats for key constant names emerging within this package. Above we have `AnnotationKeyPropagateFromName`, and here we have `ExternalResourceKindTagKey`. We should probably pick one of:

1. `AnnotationKeyPropagateFromName` and `ExternalResourceTagKeyKind` (purpose then name)
2. `PropagateFromNameAnnotationKey` and `ExternalResourceKindTagKey` (name then purpose)

I prefer option 1, because it groups all the keys that are an `AnnotationKey` or an `ExternalResourceTag` lexicographically by their prefix.

Reply:
I agree. Option 1 sounds good.